tests/test_lcov.py

```diff
@@ -0,0 +1,290 @@
+                return (l*l*l)
+
+            def IsItTrue():
+                return True
+
+        """,
+        )
+
+        self.make_file(
+            "test_file.py",
+            """\
+            #!/usr/bin/env python3
+
+            from main_file import cuboid_volume
+            import unittest
+
+            class TestCuboid(unittest.TestCase):
+                def test_volume(self):
+                    self.assertAlmostEqual(cuboid_volume(2),8)
+                    self.assertAlmostEqual(cuboid_volume(1),1)
+                    self.assertAlmostEqual(cuboid_volume(0),0)
+                    self.assertAlmostEqual(cuboid_volume(5.5),166.375)
+
+        """,
+        )
+
+    def run_coverage(self):
+        """Run coverage.py on main_file.py, and create an LCOV report."""
+        cov = coverage.Coverage(source=["."])
+        self.start_import_stop(cov, "main_file")
+        return cov
+
+    def get_lcov_report_content(self):
+        """Return the content of the LCOV report for `module`."""
+        filename = "coverage.lcov"
+        line_list = []
+        with open(filename) as f:
+            for line in f:
+                line_list.append(line)
+            return line_list
+
+    def test_no_lcov_at_start(self):
+        """Preliminary test to show that there is no lcov file at start"""
+        self.assert_doesnt_exist("coverage.lcov")
+
+    def test_lcov_file_exists(self):
+        """Test that the coverage.lcov file is produced when coverage is run"""
+        self.create_initial_files()
+        cov = self.run_coverage()
+        cov.lcov_report()
+        self.assert_exists("coverage.lcov")
+
+    def test_lone_file(self):
+        """For a single file with a couple of functions, the lcov should cover
+        the function definitions themselves, but not the returns"""
+        self.make_file(
+            "main_file.py",
+            """\
+            #!/usr/bin/env python3
+
+            def cuboid_volume(l):
+                return (l*l*l)
+
+            def IsItTrue():
+                return True
+
+        """,
+        )
+        expected_result = [
+            "TN:\n",
+            "SF:main_file.py\n",
+            "DA:3,1,7URou3io0zReBkk69lEb/Q\n",
+            "DA:6,1,ilhb4KUfytxtEuClijZPlQ\n",
+            "DA:4,0,Xqj6H1iz/nsARMCAbE90ng\n",
+            "DA:7,0,LWILTcvARcydjFFyo9qM0A\n",
+            "LF:4\n",
+            "LH:2\n",
+            "end_of_record\n",
+        ]
+        cov = coverage.Coverage(source=["."])
+        self.start_import_stop(cov, "main_file")
+        cov.lcov_report()
+        actual_result = self.get_lcov_report_content()
+        assert expected_result == actual_result
+
+    def test_simple_line_coverage_two_files(self):
+        """Test that line coverage is created when coverage is run,
+        and matches the output of the file below"""
+        self.create_initial_files()
+        cov = coverage.Coverage(source=".")
+        self.start_import_stop(cov, "main_file")
```

Comment:
Shouldn't the main_file be run by the test_file, instead of directly?

Reply:
Hi there,

Is there a way to do the above? I am just about finished with making the changes, but this change in the tests is stumping me at the moment.

Cheers,

Brad

Reply:
Sorry, I'm not sure what you are asking. My point was simply that you don't need to call `start_import_stop` on both the main_file and the test_file.  test_file.py imports main_file, so just calling it on test_file will be enough, and is how coverage would normally be run.

Reply:
Thank you, it seems I misunderstood the review comment. I have amended the tests now, thank you!